common/buildcraft/builders/tile/TileFiller.java

```diff
@@ -496,7 +496,7 @@ public IBox getBox() {
     @Override
     public void setPattern(IFillerPattern pattern, IStatementParameter[] params) {
         patternStatement.set(pattern);
-        IntStream.range(0, patternStatement.maxParams).forEach(i -> patternStatement.set(i, params[i]));
+        IntStream.range(0, params.length).forEach(i -> patternStatement.set(i, params[i]));
```

Comment:
Should this be `Math.min(params.length, patternStatement.maxParams)` instead?

Reply:
Actually never mind - I've noticed a few other places where this was wrong so I'll fix those too.

Reply:
I don't think, that it should, because filler action won't have more parameters, than patternStatement.maxParamenters.

Reply:
Technically the params array comes from the gate length, not the filler action's maximum number of parameters. (However buildcraft itself doesn't have any gates with more than 3 paramaters so this shouldn't actually ever happen)